worker/uniter/relation/relationer_test.go

```diff
@@ -304,23 +280,29 @@ func (s *RelationerImplicitSuite) TestImplicitRelationer(c *gc.C) {
 	apiRelUnit, err := apiRel.Unit(apiUnit)
 	c.Assert(err, jc.ErrorIsNil)
 
-	r := relation.NewRelationer(apiRelUnit, dir)
+	r := relation.NewRelationer(apiRelUnit, ops)
 	c.Assert(r, jc.Satisfies, (*relation.Relationer).IsImplicit)
 
 	// Hooks are not allowed.
-	f := func() { r.PrepareHook(hook.Info{}) }
+	f := func() { _, _ = r.PrepareHook(hook.Info{}) }
 	c.Assert(f, gc.PanicMatches, "implicit relations must not run hooks")
```

Comment:
It would be a good opportunity to get rid of panics. If we bubble up the error, only the uniter worker will shutdown (and restart).